src/main/java/com/amihaiemil/versioneye/MkVersionEye.java

```diff
@@ -88,7 +106,29 @@ public VersionEye trusted() throws IOException {
 
     @Override
     public Me me() {
+        if(this.authenticated == null) {
+            throw new IllegalStateException("No Authenticated User.");
+        }
         return new MkMe(this.server, this.authenticated);
     }
+    
+    /**
+     * Add authenticated user to the MkServer.
+     */
+    private void addAuthenticatedUserToServer() {
```

Comment:
also, after the refactoring and MkMe, this method will take as a parameter the authenticated json which comes in the ctor and will not be retained anymore